scipy/stats/tests/test_variation.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+from numpy.testing import assert_equal, assert_allclose
+import pytest
+from scipy.stats import variation
+
+
+class TestVariation:
```

Comment:
From what I can tell the `variation` method in numpy is numerically stable, using 2 pass w/sample mean calculations. So it isn't an immediate concern but would be good to add a case where `std` has potential to overflow if unstable textbook calculation is used. 

Think that is worthwhile for maintenance ease or am I being excessive if we already have a stable variance calculation method?

Reply:
Hmmm... there are cases where `np.std()` will overflow and return `inf` even though the answer can be represented with 64 bit floating point.  E.g. ideally the following should return `2.5e200`:

```
In [27]: np.std([1e200, -2e200, 2e200, 5e200])                                  
/Users/warren/mc37scipy/lib/python3.7/site-packages/numpy/core/_methods.py:195: RuntimeWarning: overflow encountered in multiply
  x = um.multiply(x, x, out=x)
Out[27]: inf
```
Because `variation` ultimately depends on `numpy.std`, it will also fail with this data:
```
In [61]: a = np.array([1e200, -2e200, 2e200, 5e200])

In [62]: variation(a)                                                           
/Users/warren/mc37scipy/lib/python3.7/site-packages/numpy/core/_methods.py:195: RuntimeWarning: overflow encountered in multiply
  x = um.multiply(x, x, out=x)
Out[62]: inf
```
The correct answer is `2.5e200/1.5e200 = 1.66666....`.

A little algebra with the formula for the variation leads to this version (for 1-d data and no handling of all the various edge-cases or `nan_policy`):

```
In [63]: def v2(a, ddof=0): 
    ...:     a = np.asarray(a) 
    ...:     m = a.mean() 
    ...:     b = a/m 
    ...:     return np.std(b, ddof=ddof) 
    ...:  
```
This version *does* handle those large values:
```
In [64]: v2(a)
Out[64]: 1.6666666666666663
```
I'll investigate this some more, and see if it is worthwhile using that version as the basis for our implementation.

Reply:
I think in this scenario you’d be maintaining precision with large stddev and large mean but would lose precision with the large mean small stddev cases because of the division by a large number. I haven’t tested this out but if my intuition is correct one thought is to branch the calculation logic based on some threshold for the mean (perhaps of the absolute values). Looking forward to see what you find in your experiments.

Reply:
This is a bit of a rabbit hole.  I think we're OK using `np.std(x)` for now, and leaving the alternative that I mentioned above for future investigation.